### PR TITLE
test(cadl): initial sdk for adp

### PR DESCRIPTION
**Pull request by archerzz**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/adp/Azure.AI.AutonomousDevelopmentPlatform/src/GlobalSuppressions.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Diagnostics.CodeAnalysis;
+
+[assembly: SuppressMessage("Usage", "AZC0002:DO ensure all service methods, both asynchronous and synchronous, take an optional CancellationToken parameter called cancellationToken.", Justification = "CancellationToken can be passed through RequestOptions")]
```
Comment (m-nash) [inline sdk/adp/Azure.AI.AutonomousDevelopmentPlatform/src/GlobalSuppressions.cs:6]: FYI https://github.com/Azure/azure-sdk-tools/issues/4715

File: sdk/adp/Azure.AI.AutonomousDevelopmentPlatform/src/properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (m-nash) [inline sdk/adp/Azure.AI.AutonomousDevelopmentPlatform/src/properties/AssemblyInfo.cs:1]: This folder should be capital `Properties`

Comment (archerzz) [reply sdk/adp/Azure.AI.AutonomousDevelopmentPlatform/src/properties/AssemblyInfo.cs]: done

File: sdk/adp/Azure.AI.AutonomousDevelopmentPlatform/src/GlobalSuppressions.cs (hunk 1 above)
Comment (archerzz) [reply sdk/adp/Azure.AI.AutonomousDevelopmentPlatform/src/GlobalSuppressions.cs]: @m-nash I think the global suppression here is unnecessary. I've posted details in the issue above.

**Comment by archerzz** [comment]
corresponding API view: https://apiview.dev/Assemblies/Review/f0323a0a0e7b4679896fd38c2baf6ec6

**Pull request closed without merging**